### DISCUSSION TITLE
Set grouping field to be written to files

### DISCUSSION THREAD

**Discussion by aereaux** [beetbox/beets#4435] [Q&A] [answered]
Hello,

Is there any way I can set the `grouping` field to be written programmatically, more or less to be equivalent to the `parentwork` field?  I tried (with the inline plugin):
```
item_fields:
  grouping: "parentwork"
```
But this causes `RecursionError: maximum recursion depth exceeded`.

Thanks.

**Comment by sampsyo** [comment, marked as answer]
Interesting; seeing the full traceback could help explain why this particular error is happening!

Probably the "right way" to do this would be using the "formatted modify" feature proposed in #4095. That would let you actually copy the value from one field to another, if that suits your use case.

**Reply by aereaux** [reply]
Here's the full traceback, it's a little long: [log.txt](https://github.com/beetbox/beets/files/9305168/log.txt)

Sounds like that feature would be useful, but would require me to rerun it every time a value changed.  That's one of the reasons I like the inline plugin, although it can be a little slow.

**Reply by aereaux** [reply]
Looking at this traceback more it looks like it's because I'm overriding a nonflexible attribute, and the inline plugin gets a list of all nonflexible attributes to give to my function, so infinite recursion.  Any idea on the best way to fix this?

**Reply by aereaux** [reply]
After some misadventures with plugin listeners I think I have what I want with a `template_func`.  Still not sure on the best way to fix that recursion in the inline plugin, but I'll see if I can fix that as well.

Thanks!
